tests/monotouch-test/UIKit/UIContentSizeCategoryTest.cs

```diff
@@ -0,0 +1,38 @@
+//
+// Unit tests for UIContentSizeCategory
+//
+// Authors:
+//	Vincent Dondain <vidondai@microsoft.com>
+//
+// Copyright 2017 Microsoft. All rights reserved.
+//
+
+using System;
+using Foundation;
+using NUnit.Framework;
+using UIKit;
+
+namespace MonoTouchFixtures.UIKit {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class UIContentSizeCategoryTest {
+
+		[Test]
+		public void IsAccessibilityCategory ()
+		{
+			var isAccessible = UIContentSizeCategory.AccessibilityMedium.IsAccessibilityCategory ();
+			Assert.IsTrue (isAccessible, "AccessibilityMedium");
+			isAccessible = UIContentSizeCategory.Medium.IsAccessibilityCategory ();
+			Assert.IsFalse (isAccessible, "Medium");
+		}
+
+		[Test]
+		public void Compare ()
+		{
+			var small = UIContentSizeCategory.Small;
+			var large = UIContentSizeCategory.Large;
+			Assert.True (UIContentSizeCategoryExtensions.Compare (small, large) == NSComparisonResult.Ascending, "small < large");
```

Comment:
What happens with this?

```csharp
UIContentSizeCategoryExtensions.Compare ((UIContentSizeCategory) 31415, (UIContentSizeCategory) 271828))
```

iow comparing with enum values we don't know about.

Reply:
@rolfbjarne 

```
[FAIL] UIContentSizeCategoryTest.Compare : System.NullReferenceException : Object reference not set to an instance of an object
  at UIKit.UIContentSizeCategoryExtensions.Compare (UIKit.UIContentSizeCategory category1, UIKit.UIContentSizeCategory category2) [0x00000] in /Users/vidondai/Documents/xi-master/xamarin-macios/src/UIKit/UIContentSizeCategory.cs:18 
  at MonoTouchFixtures.UIKit.UIContentSizeCategoryTest.Compare () [0x0001b] in /Users/vidondai/Documents/xi-master/xamarin-macios/tests/monotouch-test/UIKit/UIContentSizeCategoryTest.cs:44 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in /Users/vidondai/Documents/xi-master/xamarin-macios/external/mono/mcs/class/corlib/System.Reflection/MonoMethod.cs:305
```

I guess I could throw a clearer exception?

Reply:
We can always do better than a NRE :smile:

Reply:
I got this!

Reply:
@rolfbjarne done

Reply:
I don't like the solution (if it's the reflection based check from `IsDefined`).
I'm pretty sure the comparison with `nil` has a behaviour in ObjC and that we should do the same.

Reply:
@spouliot comparing to `nil` throws in ObjC:

`NSComparisonResult test = UIContentSizeCategoryCompareToCategory (UIContentSizeCategoryLarge, nil);`

```
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'UIContentSizeCategoryCompareToCategory cannot be used to order arbitrary strings, only UIContentSizeCategory objects (comparing UICTContentSizeCategoryL to (null)).'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001040f503b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000103789f41 objc_exception_throw + 48
	2   CoreFoundation                      0x00000001040fa202 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x0000000103264523 -[NSAssertionHandler handleFailureInFunction:file:lineNumber:description:] + 165
	4   UIKit                               0x00000001048651e4 UIContentSizeCategoryCompareToCategory + 430
	5   OCTest                              0x0000000102e80699 -[ViewController viewDidLoad] + 73
	6   UIKit                               0x000000010471659d -[UIViewController loadViewIfRequired] + 1235
	7   UIKit                               0x00000001047169ea -[UIViewController view] + 27
	8   UIKit                               0x00000001045e4b23 -[UIWindow addRootViewControllerViewIfPossible] + 122
	9   UIKit                               0x00000001045e522b -[UIWindow _setHidden:forced:] + 294
	10  UIKit                               0x00000001045f80cb -[UIWindow makeKeyAndVisible] + 42
	11  UIKit                               0x000000010456a561 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 4711
	12  UIKit                               0x000000010456f791 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1720
	13  UIKit                               0x0000000104934d2c __111-[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:]_block_invoke + 924
	14  UIKit                               0x0000000104d07880 +[_UICanvas _enqueuePostSettingUpdateTransactionBlock:] + 153
	15  UIKit                               0x0000000104934925 -[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:] + 249
	16  UIKit                               0x0000000104935183 -[__UICanvasLifecycleMonitor_Compatability activateEventsOnly:withContext:completion:] + 668
	17  UIKit                               0x00000001052a8e5d __82-[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:]_block_invoke + 262
	18  UIKit                               0x00000001052a8d16 -[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:] + 444
	19  UIKit                               0x0000000104f8aa62 __125-[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:]_block_invoke + 420
	20  UIKit                               0x0000000105187efe _performActionsWithDelayForTransitionContext + 100
	21  UIKit                               0x0000000104f8a85e -[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:] + 231
	22  UIKit                               0x0000000104d06edd -[_UICanvas scene:didUpdateWithDiff:transitionContext:completion:] + 392
	23  UIKit                               0x000000010456e012 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 523
	24  UIKit                               0x0000000104b428d7 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 369
	25  FrontBoardServices                  0x00000001087921b4 -[FBSSceneImpl _didCreateWithTransitionContext:completion:] + 338
	26  FrontBoardServices                  0x000000010879aca9 __56-[FBSWorkspace client:handleCreateScene:withCompletion:]_block_invoke_2 + 235
	27  libdispatch.dylib                   0x0000000107a0843c _dispatch_client_callout + 8
	28  libdispatch.dylib                   0x0000000107a0daf4 _dispatch_block_invoke_direct + 592
	29  FrontBoardServices                  0x00000001087c66ce __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 24
	30  FrontBoardServices                  0x00000001087c6384 -[FBSSerialQueue _performNext] + 464
	31  FrontBoardServices                  0x00000001087c68f3 -[FBSSerialQueue _performNextFromRunLoopSource] + 45
	32  CoreFoundation                      0x00000001040980c1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	33  CoreFoundation                      0x0000000104137cd1 __CFRunLoopDoSource0 + 81
	34  CoreFoundation                      0x000000010407c9d9 __CFRunLoopDoSources0 + 185
	35  CoreFoundation                      0x000000010407bfbf __CFRunLoopRun + 1279
	36  CoreFoundation                      0x000000010407b849 CFRunLoopRunSpecific + 409
	37  GraphicsServices                    0x00000001090bf9c6 GSEventRunModal + 62
	38  UIKit                               0x000000010457127c UIApplicationMain + 159
	39  OCTest                              0x0000000102e8078f main + 111
	40  libdyld.dylib                       0x0000000107a84d81 start + 1
	41  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Reply:
ok, let's do the same - but with a managed exception, e.g. 

```
var p1 = value1.GetHandle ();
if (p1 == IntPtr.Zero) throw new ArgumentException ($"{value1} is invalid");
...
```

That will cover the case where a constant might not be available (which would pass the slow enum check and throw the same native exception as above)

Reply:
@spouliot `GetHandle ()` are you thinking of this extension method? https://github.com/xamarin/xamarin-macios/blob/master/src/Foundation/DictionaryContainer.cs#L483-L486

It's on `DictionaryContainer`, not the case here.

Reply:
look in `src/ObjCRuntime/INativeObject.cs`

Reply:
Found in in `INativeObject` (: